apps/web/src/pages/templates/components/TemplateEditorFormProvider.tsx

```diff
@@ -164,16 +164,18 @@ const TemplateEditorFormProvider = ({ children }) => {
     }
   }, [name]);
 
-  const { template, isLoading, isCreating, isUpdating, isDeleting, updateNotificationTemplate } = useTemplateController(
-    templateId,
-    {
-      onFetchSuccess: (fetchedTemplate) => {
-        setTrigger(fetchedTemplate.triggers[0]);
-        const form = mapNotificationTemplateToForm(fetchedTemplate);
-        reset(form);
-      },
-    }
-  );
+  const { template, isLoading, isCreating, isUpdating, isDeleting, updateNotificationTemplate } =
+    useTemplateController(templateId);
+
+  useEffectOnce(() => {
+    if (!template) return;
+
+    // we don't call this on success because the templates might be fetched before
+    setTrigger(template.triggers[0]);
+    const form = mapNotificationTemplateToForm(template);
+    reset(form);
+  }, !!template);
```

Comment:
Fixed the issue I've introduced before :( basically, if the template was fetched before then this `onFetchSuccess` won't be called, and the form state won't be reset (set). Will create a separate hotfix for this.